src/test/java/org/mule/service/scheduler/internal/SchedulerThreadPoolsPerSchedulerTestCase.java

```diff
@@ -0,0 +1,139 @@
+import static org.mule.runtime.api.scheduler.SchedulerConfig.config;
+import static org.mule.tck.probe.PollingProber.DEFAULT_POLLING_INTERVAL;
+import static org.mule.test.allure.AllureConstants.SchedulerServiceFeature.SCHEDULER_SERVICE;
+
+@Feature(SCHEDULER_SERVICE)
+@RunWith(Parameterized.class)
+public class SchedulerThreadPoolsPerSchedulerTestCase extends AbstractSchedulerThreadPoolsTestCase {
+
+  @Parameterized.Parameters(name = "{0}, {2}")
+  public static Iterable<Object[]> data() {
+    List<SchedulerPoolStrategy> strategies = strategies();
+
+    Map<String, Function<SchedulerThreadPools, Scheduler>> schedulerSuppliers =
+        of("Custom",
+           service -> service.createCustomScheduler(config().withMaxConcurrentTasks(1).withDirectRunCpuLightWhenTargetBusy(true),
+                                                    CORES, () -> 1000L),
+           "CPULight", service -> service.createCpuLightScheduler(config(), CORES, () -> 1000L),
+           "CPUIntensive", service -> service.createCpuIntensiveScheduler(config(), CORES, () -> 1000L),
+           "IO", service -> service.createIoScheduler(config(), CORES, () -> 1000L));
+
+    // Combines the strategies with the schedulers.
+    List<Object[]> data = new ArrayList<>(strategies.size() * schedulerSuppliers.size());
+    for (SchedulerPoolStrategy strategy : strategies) {
+      for (Entry<String, Function<SchedulerThreadPools, Scheduler>> schedulerSupplierEntry : schedulerSuppliers.entrySet()) {
+        data.add(new Object[] {strategy, schedulerSupplierEntry.getValue(), schedulerSupplierEntry.getKey()});
+      }
+    }
+
+    return data;
+  }
+
+  protected Function<SchedulerThreadPools, Scheduler> schedulerProvider;
+  protected SchedulerPoolStrategy strategy;
+
+  public SchedulerThreadPoolsPerSchedulerTestCase(SchedulerPoolStrategy strategy,
+                                                  Function<SchedulerThreadPools, Scheduler> schedulerProvider,
+                                                  @SuppressWarnings("unused") String schedulerType) {
+    super(strategy);
+    this.schedulerProvider = schedulerProvider;
+  }
+
+  @Override
+  protected ContainerThreadPoolsConfig buildThreadPoolsConfig() throws MuleException {
+    // We want to have all pools with only one worker, so we get always the same thread for the same pool.
+    // This simplifies our assertions a lot.
+    ContainerThreadPoolsConfig threadPoolsConfig = spy(super.buildThreadPoolsConfig());
+    when(threadPoolsConfig.getUberCorePoolSize()).thenReturn(OptionalInt.of(1));
+    when(threadPoolsConfig.getUberMaxPoolSize()).thenReturn(OptionalInt.of(1));
+    when(threadPoolsConfig.getCpuLightPoolSize()).thenReturn(OptionalInt.of(1));
+    when(threadPoolsConfig.getCpuIntensivePoolSize()).thenReturn(OptionalInt.of(1));
+    when(threadPoolsConfig.getIoCorePoolSize()).thenReturn(OptionalInt.of(1));
+    when(threadPoolsConfig.getIoMaxPoolSize()).thenReturn(OptionalInt.of(1));
```

Comment:
Can't use static imports here because of conflict with ImmutableMap.of.
I thought it made it less ambiguous by having the OptionalInt here (because there is no other hint about what it is in the code), and avoid the full name on the Map one because in the same line we have the type of the variable which indicates what that "of" would be about.

But I could do it the other way around if you think it is better.